src/geofabrics/processor.py

```diff
@@ -39,24 +39,44 @@ def get_instruction_path(self, key: str) -> str:
         defaults = {'result_dem': "generated_dem.nc", 'dense_dem': "dense_dem.nc",
                     'dense_dem_extents': "dense_extents.geojson"}
 
-        if key in self.instructions['instructions']['data_paths']:
-            return self.instructions['instructions']['data_paths'][key]
-        elif "local_cache" in self.instructions['instructions']['data_paths'] and key in defaults.keys():
-            return pathlib.Path(self.instructions['instructions']['data_paths']['local_cache']) / defaults[key]
+        path_instructions = self.instructions['instructions']['data_paths']
+        local_cache = pathlib.Path(path_instructions['local_cache'])
+
+        if key == 'local_cache':
+            return local_cache
+        elif key in path_instructions:
+            # check if a list or single path
+            if type(path_instructions[key]) == list:
+                absolute_file_paths = []
+                file_paths = path_instructions[key]
+                for file_path in file_paths:
+                    file_path = pathlib.Path(file_path)
+                    file_path = file_path if file_path.is_absolute() else local_cache / file_path
+                    absolute_file_paths.append(str(file_path))
+                return absolute_file_paths
+            else:
+                file_path = pathlib.Path(path_instructions[key])
+                file_path = file_path if file_path.is_absolute() else local_cache / file_path
+                return file_path
+        elif key in defaults.keys():
+            return local_cache.absolute() / defaults[key]
         else:
```

Comment:
I find this new way with local variables much easier to read

Reply:
Awesome! I'll look at doing that elsewhere too then :) (This PR but also in the future)